pos_self_service_weight_base/static/src/js/Screens/SelfServiceScreen/WeightWidget.js

```diff
@@ -0,0 +1,49 @@
+// SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+odoo.define("pos_self_service_weight_base.WeightWidget", function (require) {
+    "use strict";
+
+    const PosComponent = require("point_of_sale.PosComponent");
+    const Registries = require("point_of_sale.Registries");
+
+    const {onMounted, onWillUnmount, useState} = owl;
+
+    class WeightWidget extends PosComponent {
```

Comment:
could this inherit from `point_of_sale.ScaleScreen` to avoid re-implementing similar methods?

Reply:
I'm not sure it's a good idea : 
- the template displays the order button : should be hidden in both tare and product mode
- the template displays the price, it should be hidden in tare mode
- it needs to be loaded with a product
- it is usually called as a popup which returns a payload

If there is a clean way to do it, please point in the right direction :-)